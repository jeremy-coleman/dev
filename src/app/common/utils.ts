import * as url from 'url';

export const uniqueId = (length?: number) => Math.random().toString(24).substr(2, length);

export function isObject(item: any): boolean {
    return (item && typeof item === 'object' && !Array.isArray(item) && item !== null);
}

export function mergeDeep(target: any, source: any): any {
    let output = Object.assign({}, target);
        {Object.keys(source).forEach(key => {
            if (isObject(source[key])) {
            if (!(key in target))
                Object.assign(output, { [key]: source[key] });
            else
                output[key] = mergeDeep(target[key], source[key]);
            } else {
            Object.assign(output, { [key]: source[key] });
            }
        });
    }
    return output;
}

export const isLocalhostUrl = (urlStr: string): boolean => {
    const parsedUrl = url.parse(urlStr);
    return (parsedUrl.hostname === 'localhost' || parsedUrl.hostname === '127.0.0.1');
}

export const isSecuretUrl = (urlStr: string): boolean => {
    const parsedUrl = url.parse(urlStr);
    return (!!parsedUrl.protocol && parsedUrl.protocol.startsWith('https'));
}

export const safeStringify = (o: any, space: string | number = undefined): string => {
 let cache = [];
 if (typeof o !== 'object')
  return `${o}`;
  return JSON.stringify(o, function (key, value) {
        if (typeof value === 'object' && value !== null) {
            if (cache.indexOf(value) !== -1) {
                return; } cache.push(value);} return value; }, space);
}

export const approximateObjectSize = (object: any, cache:any[] = []): number => {
    switch (typeof object) {
        case 'boolean': return 4;
        case 'number':  return 8;
        case 'string':  return object.length * 2;
        case 'object':  let bytes = 0; cache.push(object); for (let i in object) {let value = object[i];
            if (typeof value === 'object' && value !== null) {
            if (cache.indexOf(value) !== -1) { continue;} cache.push(value);}
                bytes += approximateObjectSize(value, cache); }
            return bytes; default: //value is null, undefined, or a function
            return 0; //this checked for infinite recursion
    }
}

export function repeat(str: string, times: number) {
    let output = '';
    for(let i=0; i<times; ++i){
        output += str;
    }
    return output;
}
