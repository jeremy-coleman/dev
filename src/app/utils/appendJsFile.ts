export function appendJsFile(url, callback) {
  var s = document.createElement('script'),
    head = document.getElementsByTagName('head')[0];
  s.type = 'text/javascript';
  s.charset = 'utf-8';
  s.src = url;
  s.onload = () => {
    callback && callback();
  };
  head.insertBefore(s, head.firstChild);
  return s;
}