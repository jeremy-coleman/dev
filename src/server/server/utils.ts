import { homedir } from 'os';
import {  join } from 'path';
import { writeJsonFile } from './fs';
import * as fs from './fs';
import * as fse from 'fs-extra';


export interface IConfig {
  port: number,
  ssl: boolean;
  sslKey: string;
  sslCert: string;
  wsPort: number;
}

export function getRootDir(): string {
  return join(homedir(), '.bstats');
}

export function getConfigPath(): string {
  return join(getRootDir(), 'config.json');
}

export function getConfig(): IConfig {
  return fse.readJsonSync(getConfigPath());
}

export function getFilePath(relativePath: string): string {
  return join(getRootDir(), relativePath);
}

export function writeInitConfig(): Promise<void> {
  return fs.exists(getConfigPath())
    .then(exists => {
      if (exists) {
        return Promise.resolve();
      } else {
        let data: IConfig = {
          port: 7200,
          ssl: false,
          sslKey: null,
          sslCert: null,
          wsPort: 7210
        };

        console.info(`initialized config file at ${getConfigPath()}`);
        return writeJsonFile(getConfigPath(), data);
      }
    });
}

export function getHumanSize(bytes: number, decimals: number = 2): string {
  if (!bytes) {
    return '0 Bytes';
  }

  const sizes: string[] = ['Bytes', 'KB', 'MB', 'GB', 'TB'];
  const k = 1000;
  const i = Math.floor(Math.log(bytes) / Math.log(k));

  return `${parseFloat((bytes / Math.pow(k, i)).toFixed(decimals))} ${sizes[i]}`;
}
