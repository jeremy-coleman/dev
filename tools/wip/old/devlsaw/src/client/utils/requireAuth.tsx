import * as React from 'react';
import { connect } from 'react-redux';
import {withRouter} from "react-router-dom";
import AuthService from '../services/auth.service'

export default function(ComposedComponent) {

    class Authenticate extends React.Component<any, any> {
        status = false;
        permissions = {
            '/': ['isLoggedIn'],
            '/dashboard': ['isLoggedIn'],
            '/notebook': ['isLoggedIn'],
            '/messaging': ['isLoggedIn',],
            '/profile': ['isLoggedIn',],
            '/search': ['isLoggedIn'],
        };

          /*
            '/messaging': ['isLoggedIn', 'isNotGuest'],
            '/profile': ['isLoggedIn', 'isNotGuest'],
            '/search': ['isLoggedIn', 'isNotGuest'],
            
            */

        componentWillMount() {
            this.status = false;
            const permissions = this.permissions[this.props.location.pathname];
            let pass = [];
            for (let i = permissions.length - 1; i >= 0; i--) {
                pass.push(new AuthService()[permissions[i]]());
            }

            if(Object.keys(pass).every(function(k){ return pass[k] })) {
                this.status = true;
            } else {
                this.props.history.push('/');
                this.status = false;
            }

        }

        componentWillUpdate(nextProps) {
            if (!nextProps.isLoggedIn) {
                this.props.history.push('/')
            }
        }

        render() {
            if(this.status){
                return (
                    <ComposedComponent />
                );
            }else {
                return (
                    <div/>
                );

            }
        }
    }

    function mapStateToProps(state) {
        return {
            isLoggedIn: state.auth.isLoggedIn
        };
    }

    return withRouter(connect(mapStateToProps)(Authenticate));
}
