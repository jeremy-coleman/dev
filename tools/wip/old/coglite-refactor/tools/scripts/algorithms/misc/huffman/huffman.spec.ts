import { huffman, Frequency } from './huffman';
import { toSerializableTree } from '../../utils';

describe('huffman', () => {
  it('should return the tree of prefix codes', () => {
    const frequences: Frequency[] = [
      { char: 'a', frequency: 45 },
      { char: 'b', frequency: 13 },
      { char: 'c', frequency: 12 },
      { char: 'd', frequency: 16 },
      { char: 'e', frequency: 9 },
      { char: 'f', frequency: 5 },
    ];
    const tree = huffman(frequences);

    // Convert the Huffman tree
    const serializable = toSerializableTree(tree, node => {
      const key = node.char ? `${node.char}:${node.frequency}` : node.frequency;

      return { key, left: node.left, right: node.right };
    });

    expect(tree).toMatchSnapshot('Huffman coding tree');
  });
});
