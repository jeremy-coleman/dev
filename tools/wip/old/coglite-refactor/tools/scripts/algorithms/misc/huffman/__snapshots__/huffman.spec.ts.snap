// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Huffman coding tree 1`] = `
│           ┌── d:16
│       ┌── 30
│       │   │   ┌── e:9
│       │   └── 14
│       │       └── f:5
│   ┌── 55
│   │   │   ┌── b:13
│   │   └── 25
│   │       └── c:12
└── 100
    └── a:45

`;
